### PR TITLE
Created welcome page widget, show at startup in Foedag

### PR DESCRIPTION
**Pull request by TarasSydor**
Given PR creates and uses WelcomePage in FOEDAG application. It contains:
- the widget itself;
- resource files for the widget, placed in etc/welcome_page/Logo.png/WelcomeDescription.txt;
- changes in root CMakeLists.txt to make sure resources are placed in application/specific folder (build/share/appName/etc)
- changes in MainWindow, so that the welcome page is shown at startup if no project is opened;
- New Project dialog reset after pressing Cancel.

The page itself looks as follows (remarks are welcome):
image
We should not hurry merging it to Raptor and Foedag-rs, as we are still waiting for logo and description.

Test with --script cmdline argument showed that welcome page is still shown, even though underneath compilation is on-going. This issue will be fixed in following commits.

### CODE REVIEW

**Comment by TarasSydor** [comment]
One more thought: might we want to move all the text to horizontal center, should we keep a group box then?

**Comment by ravic-rs** [comment]
> One more thought: might we want to move all the text to horizontal center, should we keep a group box then?

Yes, let's move it to center for now.

File: CMakeLists.txt [hunk 1]
```diff
@@ -534,6 +534,12 @@ add_custom_command(TARGET foedag-bin POST_BUILD
                   ${PROJECT_SOURCE_DIR}/examples
                   ${CMAKE_CURRENT_BINARY_DIR}/share/foedag/examples/)
 
+add_custom_command(TARGET foedag-bin POST_BUILD
```
Comment (KochynVolodymyr) [inline CMakeLists.txt:537]: please check install command. maybe need to add something for install.

File: src/MainWindow/welcomepagewidget.h [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+class QVBoxLayout;
+class QPixmap;
+class QPushButton;
+
+namespace FOEDAG {
+class WelcomePageWidget final : public QWidget {
+ public:
+  explicit WelcomePageWidget(const QString &header, const QString &sourcesPath,
+                             QWidget *parent = nullptr);
+  virtual ~WelcomePageWidget();
```
Comment (KochynVolodymyr) [inline src/MainWindow/welcomepagewidget.h:16]: this can be removed.

File: src/MainWindow/welcomepagewidget.h [hunk 3]
```diff
@@ -0,0 +1,32 @@
+#ifndef WELCOMEPAGEWIDGET_H
```
Comment (KochynVolodymyr) [inline src/MainWindow/welcomepagewidget.h:1]: copyright is missing

File: src/MainWindow/welcomepagewidget.h [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+#define WELCOMEPAGEWIDGET_H
```
Comment (KochynVolodymyr) [inline src/MainWindow/welcomepagewidget.h:2]: I would recommend to use #pragma once

File: src/MainWindow/welcomepagewidget.cpp [hunk 5]
```diff
@@ -0,0 +1,104 @@
+#include "welcomepagewidget.h"
```
Comment (KochynVolodymyr) [inline src/MainWindow/welcomepagewidget.cpp:1]: copyright is missing

File: src/MainWindow/welcomepagewidget.cpp [hunk 6]
```diff
@@ -0,0 +1,124 @@
+/*
```
Comment (ravic-rs) [inline src/MainWindow/welcomepagewidget.cpp:1]: What's the reason to create a widget within the mainwindow but not by itself like others ?

File: src/MainWindow/welcomepagewidget.cpp [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+This program is distributed in the hope that it will be useful,
+but WITHOUT ANY WARRANTY; without even the implied warranty of
+MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License
+along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include "welcomepagewidget.h"
```
Comment (ravic-rs) [inline src/MainWindow/welcomepagewidget.cpp:22]: Please use CamelCase for filenames, e.g. https://github.com/os-fpga/FOEDAG/tree/main/src/PinAssignment

File: src/MainWindow/welcomepagewidget.cpp [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+namespace {
+static constexpr auto HEADER_MARGIN = 6;
+static constexpr auto HEADER_POINTSIZE = 24;
+static constexpr auto DESCRIPTION_POINTSIZE = 14;
+static constexpr auto PAGE_MARGIN = 30;
+static constexpr auto PAGE_SPACING = 20;
+
+static const auto LOGO_PATH = QString("/etc/welcome_page/Logo.png");
+static const auto DESCRIPTION_PATH =
+    QString("/etc/welcome_page/WelcomeDescription.txt");
```
Comment (ravic-rs) [inline src/MainWindow/welcomepagewidget.cpp:43]: Update paths for windows.

**Review by alaindargelas — COMMENTED** [review]
Please make sure that if the GUI is starting with a Tcl script --script <script> we don't land on this page, but on the main editor/console/task Tab page.

File: src/MainWindow/main_window.cpp [hunk 9]
```diff
@@ -371,7 +371,12 @@ void MainWindow::createActions() {
   });
 }
 
-void MainWindow::gui_start() { showWelcomePage(); }
+void MainWindow::gui_start() {
+  if (GlobalSession->CmdLine()->Script().empty())
+    showWelcomePage();
```
Comment (ravic-rs) [inline src/MainWindow/main_window.cpp:376]: Can we also record the user selection to disable it on start-up if one chooses too? We can do this separately.

**Review by ravic-rs — COMMENTED** [review]
Comment about user disabling/enabling the landing page as needed.

**Comment by ravic-rs** [comment]
Good question, option 2 and 3. I don’t see why one would need option 1 via cmdline.

________________________________

**Comment by TarasSydor** [comment]
The most recent commits adds:
- Closing the welcome page on ESC press;
- Removes the icon from Foedag; centers it whenever it's available;
- Moves welcome page actions (with title) to the left, as it was initially;
- Adds a checkbox to not show welcome page at all.

I updated the screenshot in the PR.

File: src/MainWindow/main_window.cpp (hunk 9 above)
Comment (ravic-rs) [reply src/MainWindow/main_window.cpp:381]: @TarasSydor is this resolved ?

**Comment by ravic-rs** [comment]
@TarasSydor there's a conflict to resolve. Also, did we verify it doesn't interfere with the GUI's batch mode ?

**Comment by TarasSydor** [comment]
I tested it with a following command:
--batch --script '/home/work/Work/FOEDAG-Taras/tests/Testcases/aes_decrypt_fpga/aes_decrypt.tcl'

works fine.

**Pull request merged**